### ISSUE TITLE
Enhance Joi.strip() types

### ISSUE THREAD

**Issue by thomasmikava** [hjkcai/typesafe-joi#24]
Here is my proposal of enhancing Joi.strip()

Since strip() is used only for sanitizing purposes, typesafe-joi can handle it just like forbidden(). So the type definitions of strip() and forbidden() can be same.
In this case, 
```ts
const UserSchema = Joi.object({
    firstname: Joi.string().required(),
    lastname: Joi.string().required(),
});
const PostUser = UserSchema.keys({
    lastname: Joi.strip(),
});
```
generated type from PostUser is be 
```ts
{
    firstname: string;
    lastname: never;
}
``` 
But if we change type definitions of strip and make it behave like forbidden, generated type from PostUser will be 
```ts
{
    firstname: string;
}
``` 
which is more accurate.

P.S. I know that this project is not maintained, but please consider this enhancement. It will only take a mere minute to replace type definitions of strip() with same as forbidden()

Thanks!

**Comment by hjkcai** [comment]
Could you please help me submit a PR?

**Comment by hjkcai** [comment]
Released #25 in v2.1.0
